woocommerce-gutenberg-products-block.php

```diff
@@ -287,6 +287,37 @@ function woocommerce_blocks_plugin_outdated_notice() {
 
 add_action( 'admin_notices', 'woocommerce_blocks_plugin_outdated_notice' );
 
-// Include the Interactivity API.
-require_once __DIR__ . '/src/Interactivity/woo-directives.php';
+/**
+ * Check if Guntenberg plugin is enabled (required by the Interactivity API).
+ *
+ * This is a temporary fix until `WP_HTML_Tag_Processor` is released in WP 6.2.
+ *
+ * @param bool $enabled True if the Interactivity API is manually enabled.
+ * @return bool True if _also_ the Gutenberg plugin is enabled.
+ */
+function woocommerce_blocks_is_gutenberg_enabled( $enabled ) {
```

Comment:
Heads-up that WC core has a policy of L-2 support for WP, so we won't be able to remove this even when WP 6.2 is released.

I wonder whether it would be better to check the WP and GB versions directly instead of simply checking if GB is enabled. What do you think?

I also wonder if we could make this function generic so it's not coupled to the Interactivity API. That might be useful for other blocks. For example, [we have at least one other block that checks if GB is enabled](https://github.com/woocommerce/woocommerce-blocks/blob/0984663b66514be1bb028b8daf5efcb63d31faa6/src/BlockTypes/ProductSearch.php#L141-L158), so we could try to make this util function usable in that case as well? (That last comment is not a blocker, just me thinking out loud)

Reply:
The Gutenberg plugin kind of tries to remove the complexity of version checks etc by [simply checking if the `WP_HTML_Tag_Processor` class exists, and by loading it from its own compatibility layer otherwise](https://github.com/WordPress/gutenberg/blob/237673a1db3cf58d0d184b40ed1908db46b45651/lib/load.php#L96-L101).

Is that something we can imitate here? Something like

```php
function woocommerce_blocks_has_wp_html_tag_processor( $enabled ) {
	if ( ! function_exists( 'is_plugin_active' ) ) {
		include_once ABSPATH . 'wp-admin/includes/plugin.php';
	}
	return $enabled && class_exists( 'WP_HTML_Tag_Processor' );
}
add_filter(
	'woocommerce_blocks_enable_interactivity_api',
	'woocommerce_blocks_has_wp_html_tag_processor',
	999
);
```

We'd need to make sure that the `woocommerce_blocks_enable_interactivity_api` filter runs after the GB plugin is activated (if present), but that seems feasible 🤔 

Seems kinda preferable to me over having a version check and manual imports from GB's compat layer...

Reply:
> We'd need to make sure that the `woocommerce_blocks_enable_interactivity_api` filter runs after the GB plugin is activated (if present), but that seems feasible 🤔

@ockham I'm applying that filter in the `plugins_loaded` hook. I guess that runs after GB plugin is activated?

https://github.com/woocommerce/woocommerce-blocks/blob/25af68b1406e872bafa8221797a61c24c9eec4f2/woocommerce-gutenberg-products-block.php#L323-L324

Reply:
> I'm applying that filter in the `plugins_loaded` hook. I guess that runs after GB plugin is activated?

From what I understand, yeah: Gutenberg's code should be loaded by the time [`plugins_loaded`](https://developer.wordpress.org/reference/hooks/plugins_loaded/) is triggered.

(To be sure, you can probably try with WP 6.1; disable the plugin and verify that the interactivity code isn't loaded, then re-enable the plugin and verify that the interactivity code _is_ loaded.)